### ISSUE TITLE
Test failure: Interop\StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.cmd

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/runtime#100363]
Whenever I run tests locally, the Interop\StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.cmd test fails.

```
      Interop\StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.cmd [FAIL]

        Return code:      1
        Raw output file:      C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Reports\Interop.StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.output.txt
        Raw output:
        BEGIN EXECUTION

        C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set DOTNETCLI=C:\gh\runtime\dotnet.cmd

        C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set SUPERPMI_CACHE_DIRECTORY=c:\spmi

        C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set __DOTNET_ADD_32BIT=1

        C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set __DOTNET_PREFERRED_BITNESS=32
         "C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false" -p "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerial
  ization=true"  VBByRefStrTest.dll
        System.DllNotFoundException: Unable to load DLL 'VBByRefStrNative' or one of its dependencies: The specified module could not be found. (0x8007007E)
           at VBByRefStrNative.Marshal_Ansi(String expected, String& actual, String newValue)
           at Test.TestEntryPoint() in c:\gh\runtime\src\tests\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest.cs:line 28
        Xunit.Sdk.EqualException: Assert.Equal() Failure: Values differ
        Expected: 100
        Actual:   101
           at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in /_/src/Microsoft.DotNet.XUnitAssert/src/EqualityAsserts.cs:line 168
           at Xunit.Assert.Equal[T](T expected, T actual) in /_/src/Microsoft.DotNet.XUnitAssert/src/EqualityAsserts.cs:line 90
           at __GeneratedMainWrapper.Main() in c:\gh\runtime\src\tests\Interop\StringMarshalling\VBByRefStr\XUnitWrapperGenerator\XUnitWrapperGenerator.XUnitWrapperGenerator\SimpleRunner.g.cs:line 9
        Expected: 100
        Actual: 101
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        To run the test:
        > set CORE_ROOT=C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root
        > C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.cmd
        Stack Trace:
          c:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\TestWrappers\Interop.StringMarshalling\Interop.StringMarshalling.XUnitWrapper.cs(644,0): at Interop_StringMarshalling._VBByRefStr_VBByRefStrTest_VBByRefStrTest_.
  _VBByRefStr_VBByRefStrTest_VBByRefStrTest_cmd()
             at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
             at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
        Output:

          Return code:      1
          Raw output file:      C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Reports\Interop.StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.output.txt
          Raw output:
          BEGIN EXECUTION

          C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set DOTNETCLI=C:\gh\runtime\dotnet.cmd

          C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set SUPERPMI_CACHE_DIRECTORY=c:\spmi

          C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set __DOTNET_ADD_32BIT=1

          C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest>set __DOTNET_PREFERRED_BITNESS=32
           "C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false" -p "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSeri
  alization=true"  VBByRefStrTest.dll
          System.DllNotFoundException: Unable to load DLL 'VBByRefStrNative' or one of its dependencies: The specified module could not be found. (0x8007007E)
             at VBByRefStrNative.Marshal_Ansi(String expected, String& actual, String newValue)
             at Test.TestEntryPoint() in c:\gh\runtime\src\tests\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest.cs:line 28
          Xunit.Sdk.EqualException: Assert.Equal() Failure: Values differ
          Expected: 100
          Actual:   101
             at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in /_/src/Microsoft.DotNet.XUnitAssert/src/EqualityAsserts.cs:line 168
             at Xunit.Assert.Equal[T](T expected, T actual) in /_/src/Microsoft.DotNet.XUnitAssert/src/EqualityAsserts.cs:line 90
             at __GeneratedMainWrapper.Main() in c:\gh\runtime\src\tests\Interop\StringMarshalling\VBByRefStr\XUnitWrapperGenerator\XUnitWrapperGenerator.XUnitWrapperGenerator\SimpleRunner.g.cs:line 9
          Expected: 100
          Actual: 101
          END EXECUTION - FAILED
          FAILED
          Test Harness Exitcode is : 1
To run the test:
          > set CORE_ROOT=C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root
          > C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest\VBByRefStrTest.cmd
```

I build with:
```
call build.cmd host.native+clr.runtime+clr.corelib+clr.nativecorelib+clr.tools+clr.iltools+clr.alljits+clr.spmi+libs+libs.tests -arch x64,x86,arm64 -c Release /p:ArchiveTests=true /p:NoPgoOptimize=true
call build.cmd host.native+clr.runtime+clr.corelib+clr.nativecorelib+clr.tools+clr.iltools+clr.alljits+clr.spmi                 -arch x64,x86,arm64 -c Debug,Checked -librariesConfiguration Release /p:NoPgoOptimize=true
call src\tests\build.cmd x64 checked -priority 1
```

And run tests with:
```
src\tests\run.cmd x64 checked
```

I see these dlls that appear to be the one being looked for:
```
C:\gh\runtime\artifacts\tests\coreclr\obj\windows.x64.Checked\Native\Interop\StringMarshalling\VBByRefStr\VBByRefStrNative.dll
C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\bin\VBByRefStrNative.dll
```

but it doesn't seem to make it to the test directory:
```
C:\gh\runtime>dir /b C:\gh\runtime\artifacts\tests\coreclr\windows.x64.Checked\Interop\StringMarshalling\VBByRefStr\VBByRefStrTest
TestLibrary.dll
TestLibrary.pdb
VBByRefStrTest.cmd
VBByRefStrTest.dll
VBByRefStrTest.pdb
VBByRefStrTest_ilasmroundtrip.py
```

Presumably that is the issue? That it needs to get copied by some process to the test directory?

@jkoritzinsky @dotnet/interop-contrib

**Comment by AaronRobinsonMSFT** [comment]
This is just a busted scenario. I will work on it.

**Comment by AaronRobinsonMSFT** [comment]
Actually it looks like all the StringMarshalling tests are in some indeterminate stated :(

**Comment by AaronRobinsonMSFT** [comment]
@BruceForstall I stand corrected. These tests are folded into the uber [Interop test project](https://github.com/dotnet/runtime/blob/main/src/tests/Interop/Interop.csproj). The artifacts you are referencing are not supposed to be run directly. Build the `Interop.csproj` and that will run them within itself.

**Comment by BruceForstall** [comment]
I forgot to mention that this is the case of building the tests with `set BuildAsStandalone=true` (forgot because I _always_ build this way).

**Comment by BruceForstall** [comment]
The run is via a normal `src\tests\run.cmd x64 checked` after building the tests (including Pri-1).

**Comment by AaronRobinsonMSFT** [comment]
Okay, then this is needs some clarity by @jkoritzinsky. I honestly don't think that approach should be supported. We should have a single way to run tests not a series of custom ways to execute them.

**Comment by BruceForstall** [comment]
> I honestly don't think that approach should be supported. We should have a single way to run tests not a series of custom ways to execute them.

Well, while I appreciate your stated goal, I would dispute your characterization of `BuildAsStandalone=true` as "custom" : it was an agreed upon test build & run method when the original merged test plan was originally floated.

**Comment by AaronRobinsonMSFT** [comment]
> it was an agreed upon test build & run method when the original merged test plan was originally floated.

I see. I wasn't aware of that. Sigh... well yay for more debt...

**Comment by jkoritzinsky** [comment]
Yeah we need to change how BuildAsStandalone and the merged runner assemblies interact now that we have many tests in the interop suite that are compiled directly into the merged runner. I didn't realize that we supported building the whole test infrastructure as BuildAsStandalone=true.

BuildAsStandalone (when I designed it) was meant to be used for directed testing or for manually overriding the merged runner project to not have the filtering infra, not to effectively bring back the old infrastructure of "one process per test" and skip the merged runners. That's why I introduced the build-time filtering option to enable test filtering with BuildAsStandalone.

Now that we're almost at the end of the old infra (once the NativeAOT problems are fixed), I'll try to revisit this and get us back to a cleaner design where we can actually remove the old runner infra.
